apps/api/src/auth/auth.module.ts

```diff
@@ -21,7 +23,17 @@ import { GitlabStrategy } from '@/config/oauth-strategy/gitlab/gitlab.strategy'
         algorithm: 'HS256'
       }
     }),
-    UserModule
+    UserModule,
+    ThrottlerModule.forRootAsync({
+      imports: [ConfigModule],
+      useFactory: (config: ConfigService) => [
+        {
+          ttl: seconds(config.get('THROTTLE_TTL')),
```

Comment:
Instead of using config, try with `process.env`

Reply:
Okay just a sec let me try

Reply:
Its showing some sentry issues right now is .env changed in some way?

Reply:
SENTRY_TRACES_SAMPLE_RATE=
SENTRY_PROFILES_SAMPLE_RATE= 
Asking values for these two vars

Reply:
You won't need sentry details. Can you share your e4rror in here?

Reply:
yes

Reply:
```shell
secp256k1 unavailable, reverting to browser version
/home/prakhar/Desktop/prakhar-lab/personal/keyshade/node_modules/.pnpm/zod@3.23.8/node_modules/zod/lib/types.js:160
        throw result.error;
        ^

ZodError: [
  {
    "validation": "url",
    "code": "invalid_string",
    "message": "SENTRY_DSN must be a valid URL",
    "path": [
      "SENTRY_DSN"
    ]
  },
  {
    "code": "invalid_type",
    "expected": "number",
    "received": "nan",
    "path": [
      "SENTRY_TRACES_SAMPLE_RATE"
    ],
    "message": "Expected number, received nan"
  },
  {
    "code": "invalid_type",
    "expected": "number",
    "received": "nan",
    "path": [
      "SENTRY_PROFILES_SAMPLE_RATE"
    ],
    "message": "Expected number, received nan"
  }
]
    at get error [as error] (/home/prakhar/Desktop/prakhar-lab/personal/keyshade/node_modules/.pnpm/zod@3.23.8/node_modules/zod/lib/types.js:55:31)
    at ZodDiscriminatedUnion.parse (/home/prakhar/Desktop/prakhar-lab/personal/keyshade/node_modules/.pnpm/zod@3.23.8/node_modules/zod/lib/types.js:160:22)
    at ConfigModule.forRoot (/home/prakhar/Desktop/prakhar-lab/personal/keyshade/node_modules/.pnpm/@nestjs+config@3.2.3_@nestjs+common@10.4.2_class-transformer@0.5.1_class-validator@0.14.1_ref_yyv5h2h3lgf62ln4yzjhbcgsxe/node_modules/@nestjs/config/dist/config.module.js:78:45)
    at Object.<anonymous> (/home/prakhar/Desktop/prakhar-lab/personal/keyshade/apps/api/src/app/app.module.ts:33:18)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Module._load (node:internal/modules/cjs/loader:1104:12)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18) {
  issues: [
    {
      validation: 'url',
      code: 'invalid_string',
      message: 'SENTRY_DSN must be a valid URL',
      path: [ 'SENTRY_DSN' ]
    },
    {
      code: 'invalid_type',
      expected: 'number',
      received: 'nan',
      path: [ 'SENTRY_TRACES_SAMPLE_RATE' ],
      message: 'Expected number, received nan'
    },
    {
      code: 'invalid_type',
      expected: 'number',
      received: 'nan',
      path: [ 'SENTRY_PROFILES_SAMPLE_RATE' ],
      message: 'Expected number, received nan'
    }
  ],
  addIssue: [Function (anonymous)],
  addIssues: [Function (anonymous)],
  errors: [
    {
      validation: 'url',
      code: 'invalid_string',
      message: 'SENTRY_DSN must be a valid URL',
      path: [ 'SENTRY_DSN' ]
    },
    {
      code: 'invalid_type',
      expected: 'number',
      received: 'nan',
      path: [ 'SENTRY_TRACES_SAMPLE_RATE' ],
      message: 'Expected number, received nan'
    },
    {
      code: 'invalid_type',
      expected: 'number',
      received: 'nan',
      path: [ 'SENTRY_PROFILES_SAMPLE_RATE' ],
      message: 'Expected number, received nan'
    }
  ]
}

Node.js v20.17.0

```

Reply:
Also one thing I merged the develop branch into my branch earlier to resolve the issue of merge conflict as there were commits ahead of me hope that wouldn't have caused any issue?

Reply:
can you please set `NODE_ENV=dev` in your env file? And also include it in `.env.example`

Reply:
> Also one thing I merged the develop branch into my branch earlier to resolve the issue of merge conflict as there were commits ahead of me hope that wouldn't have caused any issue?

thats fine

Reply:
This I already did supposing the error was due to some environment issues I set the NODE_ENV in the .env but that sentry errors are still persistent

Reply:
![image](https://github.com/user-attachments/assets/09065814-6f12-4c1d-b07f-760998aa88b9)
These are added to dev schema of zod wouldn't that be causing issue ?

Reply:
> can you please set `NODE_ENV=dev` in your env file? And also include it in `.env.example`

That I did already because these things weren't working so i thought might be caused

Reply:
You should remove the sentry fields from your .env file. Leaving them blank tells zod that it does have a value. That's what is causing the issue.

Reply:
I removed them but error was still not solving so for the time being I have mocked the providers and properties causing issues in tests to default/empty for throttler module. Sort of a hack but the throttler endpoint is working so might work for now.But can fix the tests sooner or later.